### PR TITLE
Fixed all struct names, updated TOMBPC.DAT documentation

### PR DESCRIPTION
**Pull request by zdimension**

### CODE REVIEW

**Comment by TeslaRus** [comment]
lgtm

File: doc/scripting_tr2_tr3.asc [hunk 1]
```diff
@@ -129,18 +130,18 @@ Script Flags
...
+    CheatModeCheckDisabled =   (1 << 2),  // If set, it indicates that the game does not look for the cheat sequence keystrokes and events. 
+    NoInputTimeout =           (1 << 3),  // If set, it indicates that the game waits forever if there is no input (won't enter demo mode).
+    LoadSaveDisabled =         (1 << 4),  // If set, it indicates that the game does not allow save games.
+    ScreenSizingDisabled =     (1 << 5),  // If set, it indicates that the game does not allow screen resizing (with the function keys).
+    LockoutOptionRing =        (1 << 6),  // If set, it indicates that the user has no access to the Option Ring while playing the game. 
+    DozyCheatEnabled =         (1 << 7),  // If set, it indicates that the game has the DOZY cheat enabled (not present in final versions). 
+    UseXor =                   (1 << 8),  // If set, it indicates that a cypher byte was used to encrypt the strings in the script file, and is stored in the XorKey field. 
+    GymEnabled =               (1 << 9),  // Is Gym available on title screen.
+    SelectAnyLevel =           (1 << 10), // If set, it indicates that the names of the levels are all listed in the Passport.
```
Comment (Gh0stBlade) [inline doc/scripting_tr2_tr3.asc]: Ambiguous, this just enables level select when New Game is selected at the title screen in the passport.

File: doc/scripting_tr2_tr3.asc [hunk 2]
```diff
@@ -178,60 +179,62 @@ struct SequenceInfo
...
-    OP_CUTANGLE = 16,          // Associated with cutscenes; a viewpoint control? (one operand?)
-    OP_NOFLOOR = 17,           // For Floating Islands/Thames Wharf? (one operand?)
-    OP_ADDTOINVENTORY = 18,    // Give item; operand is item type (see below)
-    OP_LARASTARTANIMATION = 19,// Item-type 12 state to start level in: operand is state number
-    OP_NUMSECRETS = 20,        // Number of secrets (overrides engine's hardcoded count of them?): operand is that number
-    OP_KILLTOCOMPLETE = 21,    // Probably for Ice Palace boss (no operands)
-    OP_REMOVEAMMO = 22         // Lose your ammo and medipacks? (no operands?)
+    OP_PICTURE = 0,             // Unused. Compiles but does not show in-game. Maybe PSX. Operand is picture ID.
+    OP_PSX_START = 1,           // Unused. Does not compile. Maybe PSX. Operand is track ID.
+    OP_PSX_FMV = 2,             // Unused. Does not compile. Maybe PSX. Operand is FMV ID.
```
Comment (Gh0stBlade) [inline doc/scripting_tr2_tr3.asc]: What does not compile? the script? Did you try hex editing?

File: doc/scripting_tr2_tr3.asc [hunk 3]
```diff
@@ -178,60 +179,62 @@ struct SequenceInfo
...
+    OP_FMV = 3,                 // Display Full Motion Video. Operand is FMV ID.
```
Comment (Gh0stBlade) [inline doc/scripting_tr2_tr3.asc]: In my opinion it's probably better to change all these constants back to their original names.

1. These should match 1:1 with what is used in OpenTomb's code IIRC.
2. They're now more ambiguous and potentially hide information about sequencing systems which were used.
3. These consts var names are from the original Gameflow tool.

File: doc/scripting_tr2_tr3.asc [hunk 4]
```diff
@@ -178,60 +179,62 @@ struct SequenceInfo
...
+    OP_LEVEL = 4,               // Start a playable level. Operand is level ID.
+    OP_CINE = 5,                // Display cut scene sequence. Operand is cutscene ID.
+    OP_COMPLETE = 6,            // Display level-completion statistics panel.
+    OP_DEMO = 7,                // Display demo sequence. Operand is demo level ID.
+    OP_PSX_DEMO = 8,            // Unused. Does not compile. Maybe PSX. Operand is demo level ID.
```
Comment (Gh0stBlade) [inline doc/scripting_tr2_tr3.asc]: No, 8 = OP_JUMPTOSEQUENCE  explicitly allows jumping from one sequence to another.

File: doc/scripting_tr2_tr3.asc [hunk 5]
```diff
@@ -14,98 +14,99 @@ of flexibility.
...
 
 #if PSX // If PSX
-    uint16_t PSXStringOffsets[80+1];//Fixed at 80?
-    StringArray PSXStrings[80+1];//Fixed at 80? Usually Encrypted
+    uint16_t PSXStringOffsets[size + 1]; // size is 79 for the TR2 beta, 80 for all other versions
+    StringArray PSXStrings[size];
 #else // PC
-    uint16_t PCStringOffsets[80+1];//Fixed at 80?
-    StringArray PCStrings[80+1];//Fixed at 80? Usually Encrypted
+    uint16_t PCStringOffsets[41 + 1];
```
Comment (Gh0stBlade) [inline doc/scripting_tr2_tr3.asc]: I'm fairly certain this should not be 41+1. Size is possible to calculate with hacks as well (unrelated).

**Comment by Gh0stBlade** [comment]
Pull request must be reconsidered before merging.

File: doc/scripting_tr2_tr3.asc [hunk 6]
```diff
@@ -129,18 +130,18 @@ Script Flags
...
-    SelectAnyLevel =           (1 << 10),//(=1) Unlocks all levels
-    EnableCheatCode =          (1 << 11) //(=1) Related to in-game cheats?
+    DemoVersion =              (1 << 0),  // If set, it indicates that the game is a demo distribution.
+    TitleDisabled =            (1 << 1),  // If set, it indicates that the game has no Title Screen.
+    CheatModeCheckDisabled =   (1 << 2),  // If set, it indicates that the game does not look for the cheat sequence keystrokes and events. 
+    NoInputTimeout =           (1 << 3),  // If set, it indicates that the game waits forever if there is no input (won't enter demo mode).
+    LoadSaveDisabled =         (1 << 4),  // If set, it indicates that the game does not allow save games.
+    ScreenSizingDisabled =     (1 << 5),  // If set, it indicates that the game does not allow screen resizing (with the function keys).
+    LockoutOptionRing =        (1 << 6),  // If set, it indicates that the user has no access to the Option Ring while playing the game. 
+    DozyCheatEnabled =         (1 << 7),  // If set, it indicates that the game has the DOZY cheat enabled (not present in final versions).
```
Comment (zdimension) [inline doc/scripting_tr2_tr3.asc]: Which version? I've got the USA one and it's not enabled.

Comment (Gh0stBlade) [inline doc/scripting_tr2_tr3.asc]: If my memory serves me correctly, the bit that enables DOZY for TR2 on the PS1 isn't the same as the bit set to enable it for PC.

Comment (zdimension) [inline doc/scripting_tr2_tr3.asc]: On my PC version, bits 8 (use XOR) and 9 (Enable gym) are set, same for the PSX version. These are the original scripts. Also, I downloaded some TR2 PSX betas/demos from the TR beta compilation thread and none of these have that bit enabled.

**Comment by zdimension** [comment]
Fixed.

**Comment by zdimension** [comment]
The last 2 commits fix all broken / incorrect (e.g. tr / structures names (I used a little script that showed all broken struct anchor references).
Also, I've noticed some oddities in the naming of some structs, e.g. `tr_model` (name used in struct definition) instead of `tr_moveable` (name used in the level format specs, and in vt loader).

**Comment by Lwmte** [comment]
If I remember correctly, it was me who renamed moveable to model. I never was fond of naming hell in original TRS and all unofficial tools, cause there always was misunderstanding between moveable, item, object, etc. We now have word entity for naming certain items in level, and word model for entity type.

**Comment by zdimension** [comment]
So do you think all struct references should be links ?

**Comment by Lwmte** [comment]
Well... If they're referenced anywhere else, then yes, absolutely!

**Comment by Lwmte** [comment]
Is there any progress on this pull request? I'd like it to be merged and then separate TRS3 into its own repo :)

**Comment by zdimension** [comment]
Actually, I did everything I wanted to do, and everything that @Gh0stBlade wanted too (fixed the TOMBPC.DAT stuff), so I'm still waiting for it to be merged.

**Comment by vvs-** [comment]
This PR has conflicts which prevent it from being merged through web interface.

**Comment by zdimension** [comment]
The only conflicting file is the HTML file, so it should possible to merge everything else, and then re-generate it using Asciidoc.

**Pull request merged**